src/utils/twoFactor.js

```diff
@@ -188,34 +113,100 @@ export class TwoFactor {
         const actions = [
             ...fak2lak.map((pk) => deleteKey(pk)),
-            ...fak2lak.map((pk) => addKey(pk, functionCallAccessKey(accountId, METHOD_NAMES_LAK, null))),
-            addKey(confirmOnlyKey, functionCallAccessKey(accountId, METHOD_NAMES_CONFIRM, null)),
+            ...fak2lak.map((pk) => addKey(pk, functionCallAccessKey(accountId, METHOD_NAMES_LAK, ALLOWANCE_2FA))),
+            addKey(confirmOnlyKey, functionCallAccessKey(accountId, METHOD_NAMES_CONFIRM, ALLOWANCE_2FA)),
             deployContract(contractBytes),
-            functionCall('new', newArgs, LAK_ALLOWANCE, '0'),
+            functionCall('new', newArgs, GAS_2FA, '0'),
         ]
         console.log('deploying multisig contract for', accountId)
         return await account.signAndSendTransaction(accountId, actions);
     }
+
+    async rotateKeys(addPublicKey, removePublicKey) {
+        const { accountId } = this.getAccount()
+        return await this.signAndSendTransaction(accountId, [
+            addKey(addPublicKey, functionCallAccessKey(accountId, METHOD_NAMES_LAK, ALLOWANCE_2FA)),
+            deleteKey(removePublicKey)
+        ])
+    }
+
+    async signAndSendTransaction(receiverId, actions) {
+        const account = this.getAccount()
+        const { accountId } = account
+        const contract = getContract(account)
+        if (actions[0]?.functionCall?.methodName === 'delete_request') {
+            return await super.signAndSendTransaction(accountId, actions)
+        }
+        await deleteUnconfirmedRequests(contract)
+        const requestId = await contract.get_request_nonce()
+        const args = new Uint8Array(new TextEncoder().encode(JSON.stringify({
+            request: {
+                receiver_id: receiverId,
+                actions: convertActions(actions, accountId, receiverId)
+            }
+        })));
+        try {
+            await super.signAndSendTransaction(accountId, [
+                functionCall('add_request_and_confirm', args, GAS_2FA, '0')
+            ])
+            await this.sendRequest(requestId)
+        } catch (e) {
+            throw new WalletError('Error creating request')
+        }
+    }
+
+    async signAndSendTransactions(transactions) {
+        for (let { receiverId, actions } of transactions) {
+            await this.signAndSendTransaction(receiverId, actions)
+        }
+    }
 }
 
-const getContract = (account, accountId) => {
+const convertActions = (actions, accountId, receiverId) => actions.map((a) => {
+    const type = a.enum
+    const { gas, publicKey, methodName, args } = a[type]
+    const action = {
+        ...a[type],
+        type: type[0].toUpperCase() + type.substr(1),
+        gas: (gas && gas.toString()) || undefined,
+        public_key: (publicKey && convertPKForContract(publicKey)) || undefined,
+        method_name: methodName,
+        args: (args && Buffer.from(args).toString('base64')) || undefined,
+    }
+    delete action.publicKey
```

Comment:
should we just list all desired fields explicitly instead of copying all and then deleting?

Reply:
cleaner...